Bottom: 28a3f3e623ee22bd69b4c6b2871a7e76c2e80dcb
Top:    d406f0b85f648b725edd90718dd97ec330d172fa
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 05:23:51 +0100

Add profiling to file load






---

diff --git a/src/file.c b/src/file.c
index 724904c..35f00c6 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,10 +350,14 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
 
   /* Hack, load the mapping file */
   RefdesMapInit ();
 
+  start = clock();
+
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
     {
@@ -383,6 +387,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
